packages/app/src/navigation/SidebarNavigationContainer.vue

```diff
@@ -1,20 +1,39 @@
 <template>
   <SidebarNavigation
     :gql="query.data.value"
-    :is-loading="query.fetching.value"
+    :is-loading="isLoading"
   />
 </template>
 
 <script lang="ts" setup>
 import SidebarNavigation from './SidebarNavigation.vue'
 import { gql, useQuery } from '@urql/vue'
 import { SideBarNavigationContainerDocument } from '../generated/graphql'
+import { useRelevantRun } from '@packages/app/src/composables/useRelevantRun'
+import { ref, watchEffect, computed } from 'vue'
 
 gql`
-query SideBarNavigationContainer {
+query SideBarNavigationContainer($runNumber: Int!, $hasCurrentRun: Boolean!) {
   ...SidebarNavigation
 }
 `
 
-const query = useQuery({ query: SideBarNavigationContainerDocument })
+const relevantRuns = useRelevantRun()
+
+const variables = ref({ runNumber: -1, hasCurrentRun: false })
+
+const query = useQuery({ query: SideBarNavigationContainerDocument, variables, pause: true })
+
+const hasLoadedFirstTime = ref(false)
+
+const isLoading = computed(() => {
+  return !hasLoadedFirstTime.value || query.fetching.value
+})
+
+watchEffect(() => {
```

Comment:
We shouldn't use `watchEffect` here, as `query` is reactive and you don't want this re-executing when query updates. This should probably just be `watch(relevantRuns, () => {...})`

Reply:
I believe this is okay as written. I am pretty sure the `query` itself is not reactive and does not change here.  It is set just the one time as the result of `useQuery`.  The underlying fields of query, like `data`, are reactive.  In this case, I want the `query.executeQuery()` to run if the `variables` change.